cli/flox/src/commands/install.rs

```diff
@@ -218,20 +233,97 @@ impl Install {
         );
 
         match err {
+            EnvironmentError::Core(CoreEnvironmentError::LockedManifest(
+                LockedManifestError::ResolutionFailed(failures),
+            )) if failures.0.iter().all(|f| {
+                matches!(f, ResolutionFailure::PackageUnavailableOnSomeSystems { .. })
+            }) =>
+            {
+                let mut packages = packages
+                    .iter()
+                    .cloned()
+                    .map(|p| (p.id().to_string(), p))
+                    .collect::<HashMap<_, _>>();
+
+                for failure in &failures.0 {
+                    let ResolutionFailure::PackageUnavailableOnSomeSystems {
+                        catalog_message:
+                            MsgAttrPathNotFoundNotFoundForAllSystems {
+                                install_id,
+                                valid_systems,
+                                ..
+                            },
+                        invalid_systems: _,
+                    } = failure
+                    else {
+                        unreachable!("already checked that these failures are 'package unavailable on some systems'")
+                    };
+
+                    let Some(package_to_install) = packages.get_mut(install_id) else {
+                        warn!(install_id, "resolution failure for non-existent package");
+                        continue;
+                    };
+
+                    let PackageToInstall::Catalog(CatalogPackage { systems, .. }) =
+                        package_to_install
+                    else {
+                        warn!(
+                            install_id,
+                            ?package_to_install,
+                            "resolution failure for non-catalog package"
+                        );
+                        continue;
```

Comment:
nit: this should be unreachable, right?

Reply:
I think its "as unreachable" as the previous one, i.e. an unfortunate combination of resolution response and request params.
These should be very unlikely and most likely a bug, but since that sits partly in an external system i was hesitent to panic with `unreachable!` here.